### ISSUE TITLE
Use GitHub Actions deployment for GitHub Pages

### ISSUE THREAD

**Issue by TWiStErRob** [TWiStErRob/twisterrob.github.io#24]
https://github.blog/changelog/2022-07-27-github-pages-custom-github-actions-workflows-beta/

**Comment by TWiStErRob** [comment]
Can PRs be deployed?

Promising:
![image](https://github.com/TWiStErRob/twisterrob.github.io/assets/2906988/a69e45bb-2361-4772-b6f6-af5154bda6c6)

> If the workflow was triggered by a push to the default branch, use the [actions/deploy-pages](https://github.com/actions/deploy-pages) action to deploy the artifact. This step is skipped if the workflow was triggered by a pull request. -- [docs](https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site#creating-a-custom-github-actions-workflow-to-publish-your-site:~:text=This%20step%20is%20skipped%20if%20the%20workflow%20was%20triggered%20by%20a%20pull%20request.)

**Comment by ohadschn** [comment]
@TWiStErRob note this:
```ruby
# http://jekyllrb.com/docs/windows/#auto-regeneration
# For --watch to work on Windows.
gem 'wdm', '>= 0.1.1' if Gem.win_platform?
```

In theory, GitHub actions could be executed on Windows, which would try installing this Gem, resulting in that annoying warning again (about bundler not being able to satisfy the dependencies).

Here's what we did:
```ruby
install_if -> { ENV["GITHUB_ACTIONS"] != "true" } do
    puts "Is GitHub action: #{ENV["GITHUB_ACTIONS"] == "true"}"
    gem "webrick", "~> 1.8"

    install_if Gem.win_platform? do
        puts "Is Windows platform: #{Gem.win_platform?}"
        gem "wdm", "~> 0.1.0"
    end
end 
```

**Comment by TWiStErRob** [comment]
Thanks for the heads up @ohadschn, good point.

However, I think I'm going to leave it for now, as it's unlikely that it'll be run on Windows (by GitHub at least). Also, I added the condition as temporary because when I fix #24, the warning will not be relevant any more since I'll [bring my own versions](https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site#creating-a-custom-github-actions-workflow-to-publish-your-site). I just wanted to start with a clean output, so I can see clearly if something goes awry.

**Comment by TWiStErRob** [comment]
How to ensure dependency upgrades don't break site?

 * Diffed github-pages.zip by changing only README.md and the differences are a few dates in atom.xml and sitemap.xml.
 * Workflow to cache main and diff on build unless content-change label.

https://www.kenmuse.com/blog/the-many-shas-of-a-github-pull-request/
